### DISCUSSION TITLE
How to experiment with a fork?

### DISCUSSION THREAD

**Discussion by 12AT7** [wasm-bindgen/wasm-bindgen#2815] [Q&A]
I am brand new to Rust; I know a lot of C++ and Python.  This might be a FAQ somewhere, but I don't know where to look.

I want to try adding a method to `RequestInit` to set `type: 'bytes'` and attempt to get a BYOB reader.  This looks pretty easy; I just have to copy the very similar `mode()` method from `gen_RequestInit.rs`.  So to try this experiment, I forked `wasm-bindgen` and modified my Cargo.toml:
```
[dependencies]
# wasm-bindgen = { version = "0.2.79", features = ["serde-serialize"] }
wasm-bindgen = { git = "https://github.com/JohnGalbraith/wasm-bindgen", vers
ion = "0.2.79", features = ["serde-serialize"] }
```
but now I am stuck on this error:
```
❯ cargo check
    Updating git repository `https://github.com/JohnGalbraith/wasm-bindgen`
error: failed to select a version for `wasm-bindgen-shared`.
    ... required by package `wasm-bindgen-macro-support v0.2.79`
```
What other steps do I need to use to link to my own fork?

**Comment by alexcrichton** [comment]
I think you probably just need to drop the `version` annotation as the version number in git may differ from the version in your fork.
